### PR TITLE
Enhancement/updateprofile

### PR DESCRIPTION
**Pull request by SubhankarPradhan88**
Fetch UserProfile endpoint:
1. Controller method for getUser details
2. Handle Service and Dao layer
3. Handle query in entity

### CODE REVIEW

File: quora-api/src/main/java/com/upgrad/quora/api/exception/RestExceptionHandler.java [hunk 1]
```diff
@@ -50,4 +50,11 @@
                 new ErrorResponse().code(exc.getCode()).message(exc.getErrorMessage()), HttpStatus.FORBIDDEN
         );
     }
+
+    @ExceptionHandler(UserNotFoundException.class)
+    public ResponseEntity<ErrorResponse> authenticationFailedException(UserNotFoundException exc, WebRequest request) {
+        return new ResponseEntity<>(
+                new ErrorResponse().code(exc.getCode()).message(exc.getErrorMessage()), HttpStatus.UNAUTHORIZED
```
Comment (post2shyam) [inline quora-api/src/main/java/com/upgrad/quora/api/exception/RestExceptionHandler.java]: Please use appropriate HTTPStatus return code here.

File: quora-service/src/main/java/com/upgrad/quora/service/business/AdminService.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        }
+
+        UserEntity existingUser = this.userDao.getUserById(userId);
+
+        if(existingUser == null) {
+            throw new UserNotFoundException(
+                    "USR-001", "User with entered uuid to be deleted does not exist");
+        }
+
+        UserEntity deletedUser = this.userDao.deleteUser(userId);
```
Comment (post2shyam) [inline quora-service/src/main/java/com/upgrad/quora/service/business/AdminService.java]: 'this' keyword is redundant.

File: quora-service/src/main/java/com/upgrad/quora/service/dao/UserDao.java [hunk 3]
```diff
@@ -79,10 +77,10 @@ public void updateUserEntity(final UserEntity updatedUserEntity) {
     }
 
     /**
-     * Delete a user by given id from the DB.
+     * Method to delete user by id
      *
-     * @param userId Id of the user whose information is to be fetched.
-     * @return User details which is to be deleted if exist in the DB else null.
+     * @param userId : username which you want to delete
```
Comment (post2shyam) [inline quora-service/src/main/java/com/upgrad/quora/service/dao/UserDao.java]: Some places its 'userid' other places its 'uuid'. Please follow same conventions.

**Review by post2shyam — CHANGES_REQUESTED** [review]

File: quora-api/src/main/java/com/upgrad/quora/api/exception/RestExceptionHandler.java (hunk 1 above)
Comment (SubhankarPradhan88) [reply quora-api/src/main/java/com/upgrad/quora/api/exception/RestExceptionHandler.java]: I have provided NOT_FOUND HttpsStatus, for the highlighted exception.

File: quora-service/src/main/java/com/upgrad/quora/service/dao/UserDao.java (hunk 3 above)
Comment (SubhankarPradhan88) [reply quora-service/src/main/java/com/upgrad/quora/service/dao/UserDao.java]: Changed userId to uuid.

**Review by post2shyam — APPROVED** [review]

**Pull request merged**